pkg/inject/inject.go

```diff
@@ -283,7 +288,9 @@ func (conf *ResourceConfig) JSONToYAML(bytes []byte) ([]byte, error) {
 	if err := json.Unmarshal(bytes, obj); err != nil {
 		return nil, err
 	}
-	return yaml.Marshal(obj)
+
+	filteredJSON, _ := jsonfilter.Marshal(&obj)
```

Comment:
Thanks for the contribution :) 
Add a check for err.
Also, Can you move this Conversion functionality into a common function and re-use that?

Reply:
@Pothulapati Sure, will do! If this approach seems fine, I could go ahead and clean up the code a bit! Just needed a thumbs-up on this kind of implementation